torch/_refs/__init__.py

```diff
@@ -3794,7 +3799,7 @@ def _reshape_view_helper(a: TensorLikeType, *shape, allow_copy: bool) -> TensorL
             continue
 
         # Skips dimensions that are already the correct length
-        if guard_size_oblivious(length == a_.shape[idx]):
+        if guard_or_false(length == a_.shape[idx]):
```

Comment:
I guess the assumption here is that the code bellow can handle this case properly. 

here is a test to actually confirm that 
https://github.com/pytorch/pytorch/pull/150685